playbooks/roles/configure/tasks/locate_zowe_start.yml

```diff
@@ -0,0 +1,48 @@
+---
+# This task finds the zowe-start.sh in known folders
+
+# Variables:
+# - zowe_instance_dir
+# - zowe_root_dir
+# Output:
+# - zowe_start_path
+
+# validate variables
+- import_role:
+    name: common
+    tasks_from: validate_variables
+  vars:
+    variable_list:
+    - zowe_root_dir
+    - zowe_instance_dir
+
+- name: Locate zowe-start.sh
+  block:
+  - name: Init zowe_start_path variable
+    set_fact:
+      zowe_start_path: ""
+  - name: Check if zowe-start.sh is located at instanceDir
+    raw: test -f {{ zowe_instance_dir }}/bin/zowe-start.sh && echo "{{ zowe_instance_dir }}/bin/zowe-start.sh" | tr -d '\n'
+    register: zowe_start_at_instancedir
+    ignore_errors: yes
+    when: zowe_start_path == ""
+  - name: Set zowe_start_path to {{ zowe_start_at_instancedir.stdout }}
+    set_fact:
+      zowe_start_path: "{{ zowe_start_at_instancedir.stdout }}"
+    when: zowe_start_path == "" and zowe_start_at_instancedir.stdout != ''
+  - name: Check if zowe-start.sh is located at rootDir
+    raw: test -f {{ zowe_root_dir }}/scripts/zowe-start.sh && echo "{{ zowe_root_dir }}/scripts/zowe-start.sh" | tr -d '\n'
+    register: zowe_start_at_rootdir
+    ignore_errors: yes
+    when: zowe_start_path == ""
+  - name: Set zowe_start_path to {{ zowe_start_at_rootdir.stdout }}
+    set_fact:
+      zowe_start_path: "{{ zowe_start_at_rootdir.stdout }}"
+    when: zowe_start_path == "" and zowe_start_at_rootdir.stdout != ''
```

Comment:
Is this bit of code for old versions? I think zowe-start.sh has been in the instance-dir since 1.8

Reply:
@stevenhorsman Thanks for the feedback.

The main reason is trying to implement some backward compatibility, especially for uninstall process. The complicated part of uninstall is we don’t know what version is previously installed on the server, so we check all known variations. For example, check all known job names we once used and try to stop xmem server. locate-zowe-stop was part of the effort. It tries to stop Zowe normally if the user has v1.2 installed, manually or by other automation.

Originally I was trying to make the playbooks to work with all versions, like we can declare it support Zowe v1.8.0 and above. And ask the user always use the most recent version playbooks to install any Zowe version it supports. This backward compatibility actually make uninstall easier. So the user always have latest playbook knows well how to clean up Zowe.

But later I realize this may not be realistic. Using old version playbooks to install old versions could be the only reasonable choice. The complicated part of uninstall is still not fully solved, and may become even more complicated if we ask the user to use old version playbooks to install old version Zowe. The uninstall playbook exists in old version may not work well if the user has a newer versions Zowe on the server.

So all these code were part of this dilemma. But for locate-zowe-start, the code you pointed out is not useful if we do ask user to install old Zowe with old playbooks, also with the fact that playbooks only support v1.9.0+.